knowledge_repo/postprocessors/extract_images.py

```diff
@@ -13,10 +13,37 @@ class ExtractImages(KnowledgePostProcessor):
     @classmethod
     def process(cls, kp):
         images = cls.find_images(kp.read())
-        cls.collect_images(kp, images)
+        image_mapping = cls.collect_images(kp, images)
+        cls.update_thumbnail_uri(kp, images, image_mapping)
         cls.cleanup(kp)
 
     @classmethod
+    def update_thumbnail_uri(cls, kp, images, image_mapping):
+        thumbnail = kp.headers.get('thumbnail', 0)
+
+        # if thumbnail is a number, select the nth image in this post as the thumbnail
+        if isinstance(thumbnail, str) and thumbnail.isdigit():
```

Comment:
I'll handle this in a subsequent patch, since this is also an issue in several other places.